src/admin.rs

```diff
@@ -106,312 +101,549 @@ impl AdminScClient {
+        if let Some(client) = self.inner.clone() {
+            let mut client = client.write().await.admin().await;
+            client.delete::<CustomSpuSpec, _>(key.0).await?;
+            Ok(())
+        } else {
+            Err(FluvioError::Other("failed to lock client".to_owned()))
+        }
     }
 
+    #[node_bindgen]
+    async fn create_managed_spu(
+        &self,
+        name: String,
+        spec: SpuGroupSpecWrapper,
+    ) -> Result<(), FluvioError> {
+        if let Some(client) = self.inner.clone() {
+            let mut client = client.write().await.admin().await;
+            client.create::<SpuGroupSpec>(name, false, spec.0).await?;
+            Ok(())
+        } else {
+            Err(FluvioError::Other("failed to lock client".to_owned()))
+        }
+    }
 
     #[node_bindgen]
-    async fn list_spu(&self) -> Result<ArrayBuffer,ClientError> {
+    async fn delete_managed_spu(&self, name: String) -> Result<(), FluvioError> {
+        if let Some(client) = self.inner.clone() {
+            let mut client = client.write().await.admin().await;
+            client.delete::<SpuGroupSpec, _>(name).await?;
+            Ok(())
+        } else {
+            Err(FluvioError::Other("failed to lock client".to_owned()))
+        }
+    }
+}
+
+pub struct ReplicaStatusWrapper(ReplicaStatus);
 
-        debug!("list spu");
+impl TryIntoJs for ReplicaStatusWrapper {
+    fn try_to_js(self, js_env: &JsEnv) -> Result<napi_value, NjError> {
+        let mut status = JsObject::create(js_env)?;
 
-        let client = self.inner.as_ref().unwrap().clone();
-        let mut client_w = client.write().await;
+        let spu = js_env.create_int32(self.0.spu);
+        let hw = js_env.create_int64(self.0.hw);
+        let leo = js_env.create_int64(self.0.leo);
 
-        let spus = client_w.list_spu(false).await?;
+        status.set_property("spu", spu.try_to_js(js_env)?)?;
+        status.set_property("hw", hw.try_to_js(js_env)?)?;
+        status.set_property("leo", leo.try_to_js(js_env)?)?;
 
-        let json_slice = serde_json::to_vec(&spus)
-        .map_err(|err| ClientError::Other(format!("serialization error: {}",err.to_string())))?;
-       
-        Ok(ArrayBuffer::new(json_slice))
+        status.try_to_js(js_env)
     }
+}
 
+pub struct PartitionSpecWrapper(PartitionSpec);
 
+pub struct PartitionStatusWrapper(PartitionStatus);
 
-    #[node_bindgen]
-    async fn create_custom_spu(&self,param: CustomSpuParam) -> Result<(),ClientError>  {
+impl TryIntoJs for PartitionStatusWrapper {
+    fn try_to_js(self, js_env: &JsEnv) -> Result<napi_value, NjError> {
+        let mut status = JsObject::create(js_env)?;
 
-        let client = self.inner.as_ref().unwrap().clone();
-        let mut client_w = client.write().await;
+        let resolution = js_env.create_string_utf8(match self.0.resolution {
+            PartitionResolution::Offline => "Offline",
```

Comment:
Make string slice as constants

Reply:
`PartitionResolution` should implement `ToString`; I can open a PR for that; currently I am going to change this to a wrapped implementation:

I don't think creating these as const values is going to offer much better mapping than implementing a `.to_string()` method.

```rust
pub struct PartitionResolutionWrapper(PartitionResolution);

impl ToString for PartitionResolutionWrapper {
    fn to_string(&self) -> String {
        let status = match self.0 {
            PartitionResolution::Offline => "Offline",
            PartitionResolution::Online => "Online",
            PartitionResolution::LeaderOffline => "LeaderOffline",
            PartitionResolution::ElectionLeaderFound => "ElectionLeaderFound",
        };

        status.to_string()
    }
}
```